### PR TITLE
Interchain Queries Implemenation

### PR DESCRIPTION
**Pull request by jackzampolin**

### CODE REVIEW

File: docs/apps/icq/overview.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<!--
+order: 1
+-->
+
```
Comment (danburck) [inline docs/apps/icq/overview.md:4]: could add a list of prerequisite learnings, e.g. `whats an acknowledgement?`

**Review by danburck — COMMENTED** [review]
Consider splitting this PR up into smaller pieces, starting with only the specs and docs. This way we can agree on the overall implementation and it will be easier to review the following PRs.  You could split up into: 

- specs and docs
- proto and types
- keeper and module wiring

E.g. when I opened this PR I thought of: 
* PR Description is missing or link to specification can be added to provide more context
* Lifecycle of the ibc query is not very clear in the documentation

**Comment by georgelombardi97** [comment]
I will need this for a product that I am planning for. If there are no issues, would be great if we could have this merged in ibc-go so we chains will deploy it by default. Asking and convincing each chain to add this later would be hard imho.

**Comment by georgelombardi97** [comment]
Can someone please explain why this is not merged? Maybe I can help?

**Comment by adamewozniak** [comment]
+1 on prioritizing this PR ❤️

**Comment by womensrights** [comment]
We have written up our findings on Interchain / Cross-chain queries in [this discussion](https://github.com/cosmos/ibc-go/discussions/2866), please let me know your thoughts. Just to note that having a queries implementation in the ibc-go repo would not mean chains deploy it by default cc @georgelombardi97

**Pull request closed without merging**